### DISCUSSION TITLE
mypy start to fail with FastApi version 0.109.0

### DISCUSSION THREAD

**Discussion by sam-som** [fastapi/fastapi#10994] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_methods=["*"],
    allow_headers=["*"],
)
```

### Description

This block of code used to pass `mypy` test without issue.
However,after FastApi got updated to 0.109.0, it began to fail with the following error

```
main.py:23: error: Argument "allow_origins" to "add_middleware" of "Starlette" has incompatible type "list[str]"; expected <nothing>  [arg-type]
main.py:24: error: Argument "allow_methods" to "add_middleware" of "Starlette" has incompatible type "list[str]"; expected <nothing>  [arg-type]
main.py:25: error: Argument "allow_headers" to "add_middleware" of "Starlette" has incompatible type "list[str]"; expected <nothing>  [arg-type]
```

The main change in FastApi 0.109.0 is the upgrade to Starlette (`Upgrade Starlette to >=0.35.0,<0.36.0.`). 
To temporarily resolve this error, we lock `fastapi==0.108.0`. 

I'm not very familiar with `Starlette` so I'm not sure what's in `Starlette 0.35.0` that is causing the issue

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

2.4.0

### Python Version

3.10.13

**Comment by Kludex** [comment, marked as answer]
Bump mypy.

**Comment by sam-som** [comment]
Ah i see. 
We can't update mypy right now due to some dependency issue. 
We'll bump both mypy and fastapi once the dependency issue is resolve on our end.
Thank you
